doc/architecture.md

```diff
@@ -0,0 +1,136 @@
+server call the above test command will look similar to this:
+```
+/usr/local/bin/python app/adapter.py test /tmp/tmpe1iu4msr/input_pipe /tmp/tmpe1iu4msr/output_pipe
+```
+
+### Adapter Container Lifecycle
+Containers are managed by the _Collector_ process that runs on a Cloud Proxy. A new
+temporary container will be started each time 'Validate Connection' is run from the
+Account creation dialog, and when saving a new account. This container will be stopped
+once the test is complete or the new account has been created. After the account/adapter
+instance has been created, the Collector process then creates a new container. This
+container is long-lived and tied to the specific adapter instance it was created for.
+
+That is, if there are three running adapter instances for a given management pack,
+there will also be three running containers. This allows isolation between adapter
+instances, and for simultaneous collections to occur without having to guard against
+race conditions, deadlocks, etc. within the adapter's collection code.
+
+Each container will continue to exist until or unless:
+* The adapter instance is stopped or deleted
+* The adapter instance is moved to a new collector, or migrates between collectors in a
+  collector group
+* An unrecoverable error occurs in the container (such as exceeding the container's
+  memory limit)
+
+### Collection
+When performing a collection, test connection, definition, or get endpoints request,
+the following sequence takes place:
+1. VMware Aria Operations Collector sends request to adapter.
+2. Server reads `commands.cfg` to determine how to handle the request.
+3. Server starts a subprocess with the command and arguments from `commands.cfg`, plus
+   the input and output named pipe arguments.
+4. Server sends the request payload to 'input' named pipe.
+5. Server waits for subprocess to complete and write its response to the 'output' 
+   named pipe, then reads the result.
+7. Server processes the result and sends it as a response to the original REST request.
+8. If any steps failed, clean up and send an appropriate error message as REST request
+   response.
+
+Note: The `mp-test` tool by default has a timeout of 5 minutes for each REST request.
+This is different than the Cloud Proxy's `Adapter Handler`, which does not have any
+timeout. Thus, it is possible when running on a Cloud Proxy for a collection to 
+wait forever at step (5) above.
+
+### Registry
+Containerized adapters make use of an external container registry to distribute images.
+The user requires push access when building a Pack File, and the VMware Aria Operations
+Cloud Proxy needs anonymous pull access to the registry.
+
+Within a Pack File, the adapter container image is referenced by the registry location
+and image digest SHA.
+
+The following diagram shows how the processes for building and installing a Management 
+Pack work together with the registry.
+
+![Building and installing a Pack File](registry.png)
```

Comment:
There are too many things happening in this one image. Can we split it into two images, one for build, one for install?

Reply:
What I really want to show here is the through line of the pack file and container image between the two processes. Maybe there's some details that aren't relevant, but it's a fine line between too much detail and oversimplification. For now, I've just added some additional text around it and renamed the section it's in to hopefully be clearer about what the diagram is trying to accomplish